vnext/ReactUWP/Modules/AppearanceModule.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "AppearanceModule.h"
+
+#include <winrt/Windows.UI.ViewManagement.h>
+
+using Application = winrt::Windows::UI::Xaml::Application;
+using ApplicationTheme = winrt::Windows::UI::Xaml::ApplicationTheme;
+using UISettings = winrt::Windows::UI::ViewManagement::UISettings;
+
+using Method = facebook::xplat::module::CxxModule::Method;
+
+namespace react::uwp {
+
+AppearanceChangeListener::AppearanceChangeListener(std::weak_ptr<IReactInstance> &&reactInstance) noexcept
+    : Mso::ActiveObject<>(Mso::DispatchQueue::MainUIQueue()), m_weakReactInstance(std::move(reactInstance)) {}
+
+void AppearanceChangeListener::Initialize() noexcept {
+  m_currentTheme = Application::Current().RequestedTheme();
+
+  // UISettings will notify us on a background thread regardless of where we construct it or register for events. Let it
+  // be constructed before init, but redirect callbacks to the UI thread where we can check app theme.
+  m_uiSettings.ColorValuesChanged([weakThis{Mso::WeakPtr(this)}](const auto &, const auto &) noexcept {
```

Comment:
I should assign this to a revoker. I kind of assumes that death of UISettings would deregister events, but this isn't super well documented so I should do the safe thing.

Reply:
Yes, it would be nice, otherwise after several reloads we may get interesting behavior.

---
In reply to: [409749605](https://github.com/microsoft/react-native-windows/pull/4622#discussion_r409749605) [](ancestors = 409749605)